parser/SqlParser.ypp

```diff
@@ -759,6 +780,48 @@ opt_table_constraint_def_commalist:
     /* $$ = $1; */
   };
 
+opt_block_properties:
+  TOKEN_WITH TOKEN_BLOCKPROPERTIES '(' block_property_item_list ')' {
+    $$ = new quickstep::ParseBlockProperties(@1.first_line, @1.first_column, $4);
+  } 
+  | {
+    $$ = NULL;
+  };
+
+block_property_item_list:
+  block_property_item {
+    $$ = new quickstep::PtrList<quickstep::ParseBlockPropertyItem>();
+    $$->push_back($1);
+  }
+  | block_property_item_list ',' block_property_item {
+    $$ = $1;
+    $$->push_back($3);    
+  };
+
+block_property_item:
+  TOKEN_TYPE any_name {
+    $$ = quickstep::ParseBlockPropertyItem::GetTypeItem(@1.first_line, @1.first_column, $2);
+  }
+  | TOKEN_SORT any_name {
+    $$ = quickstep::ParseBlockPropertyItem::GetSortItem(@1.first_line, @1.first_column, $2);
+  }
+  | TOKEN_COMPRESS '(' name_commalist ')' {
+    $$ = quickstep::ParseBlockPropertyItem::GetCompressAttributesItem(@1.first_line, @1.first_column, $3);
+  } 
+  | TOKEN_COMPRESS TOKEN_ALL {
+    $$ = quickstep::ParseBlockPropertyItem::GetCompressAllItem(@1.first_line, @1.first_column,
+                                                               @2.first_line, @2.first_column);
+  }
+  | TOKEN_BLOCKSIZE TOKEN_UNSIGNED_NUMVAL TOKEN_MB {
+    $$ = quickstep::ParseBlockPropertyItem::GetBlockSizeItem(@1.first_line, @1.first_column, $2);
+  }
+  | TOKEN_BLOCKSIZE TOKEN_UNSIGNED_NUMVAL TOKEN_GB {
+    delete $2;
+    $$ = nullptr;
+    NotSupported(&@3, yyscanner, "Gb unit not supported");
```

Comment:
Why not? If you only want to allow units of megabytes, we might as well drop the `TOKEN_GB` entirely and only have the above case.